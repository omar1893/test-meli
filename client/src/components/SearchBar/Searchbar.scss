@import "../../../src/index.scss";

.header {
  @include grid-basic($container: true);
  background-color: $yellow-ml;
  padding-top: 0.6rem;
  padding-bottom: 0.6rem;

  .logo {
    @include grid-positioning(2, 3);
  }

  .header-searchInput {
    @include grid-positioning(3, 12, true);
    display: flex;

    input {
      border: none;
      border-radius: 0.1rem;
      font-size: 1.8rem;
      padding: 1rem;
      width: 100%;

      &::placeholder {
        color: $dark-gray;
      }
    }

    button {
        border: none;
        padding: 1rem 1.5rem;
        cursor: pointer;

        img {
            pointer-events: none;
        }
    }
  }
}
