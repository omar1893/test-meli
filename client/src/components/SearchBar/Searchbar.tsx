import React, { useState } from "react";
import "./Searchbar.scss";
import LogoMl from "../../assets/Logo_ML.png";
import SearchIcon from "../../assets/ic_Search.png";
import { useNavigate } from "react-router-dom";

const Searchbar = () => {
  const [searchValue, setSearchValue] = useState("");
  const navigate = useNavigate();

  const submitSearch = () => {
    if (searchValue !== "" && searchValue.length > 2) {
      navigate(`/items?search=${searchValue}`);
    }
  };

  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {    
    if (event.key === "Enter") {
      submitSearch();
    }
  };

  return (
    <header className="header">
      <img className="logo" src={LogoMl} alt="Logo Mercado Libre" />
      <div className="header-searchInput">
        <input
          type="text"
          value={searchValue}
          onChange={(event) => setSearchValue(event.target.value)}
          onKeyDown={(event) => handleKeyPress(event)}
          placeholder="Nunca dejes de buscar"
        />
        <button type="button" onClick={submitSearch}>
          <img src={SearchIcon} alt="Search Icon" />
        </button>
      </div>
    </header>
  );
};

export default Searchbar;