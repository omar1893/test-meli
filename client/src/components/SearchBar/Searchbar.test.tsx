import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Searchbar from "./Searchbar";
import { useNavigate } from "react-router-dom";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

describe("Searchbar", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Should match snapshot", () => {
    const { container } = render(<Searchbar />);
    expect(container).toMatchSnapshot();
  });

  it("Should navigate to search results on button click", () => {
    const navigateMock = jest.fn();
    (useNavigate as jest.Mock).mockImplementation(() => navigateMock);

    render(<Searchbar />);

    const searchInput = screen.getByPlaceholderText("Nunca dejes de buscar");
    const searchButton = screen.getByRole("button");

    fireEvent.change(searchInput, { target: { value: "example search" } });
    fireEvent.click(searchButton);

    expect(navigateMock).toHaveBeenCalledWith("/items?search=example search");
  });
});