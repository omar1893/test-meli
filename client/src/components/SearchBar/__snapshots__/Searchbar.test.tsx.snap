// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <img
      alt=""
      src="Logo_ML.png"
    />
    <div
      class="header-searchInput"
    >
      <input
        placeholder="Nunca dejes de buscar"
        type="text"
        value=""
      />
      <button
        type="button"
      >
        <img
          alt=""
          src="ic_Search.png"
        />
      </button>
    </div>
  </header>
</div>
`;
